### PR TITLE
No auth Repo In Local Gitserver

### PR DESCRIPTION
**Pull request by shubhamdixit863**
Fixes #TODO

### Modifications

Modified the Dockerfile such that git repos should have proper kubernetes resource.yaml files. As currently gitsync throws error  in BDD tests while syncing the repos as the resource yaml is invalid . 
2- Added  Provision to have repository with No authentication for BDD tests

### Verification

Tested in local kubernetes cluster

### CODE REVIEW

File: tests/gitserver/Dockerfile [hunk 1]
```diff
@@ -94,13 +114,29 @@ RUN for i in $(seq 1 $REPO_COUNT); do \
     git init && \
     git config --global user.email "you@example.com" && \
     git config --global user.name "Your Name" && \
-    echo "some: data" > data.yaml && \
+    echo "apiVersion: apps/v1\nkind: Deployment\nmetadata:\n  name: nginx-deployment\nspec:\n  selector:\n    matchLabels:\n      app: nginx\n  replicas: 2\n  template:\n    metadata:\n      labels:\n        app: nginx\n    spec:\n      containers:\n      - name: nginx\n        image: nginx:1.14.2\n        ports:\n        - containerPort: 80" > data.yaml && \
```
Comment (xdevxy) [inline tests/gitserver/Dockerfile:117]: If the BDD test need to verify syncing a Numaflow pipelines, this needs to be updated to the  pipelines/controller/isbs spec? It seems to be a bit rigid.

Comment (shubhamdixit863) [reply tests/gitserver/Dockerfile:117]: yeah bdd tests dont use this ,i will update the pr

Comment (xdevxy) [reply tests/gitserver/Dockerfile:117]: Sounds good, move to draft for now.

**Review by juliev0 — APPROVED** [review]

**Comment by juliev0** [comment]
@shubhamdixit863 should this be "Ready for Review" now? I just approved

**Comment by dpadhiar** [comment]
Is this ready to be merged? @shubhamdixit863

File: tests/gitserver/Dockerfile [hunk 2]
```diff
@@ -46,20 +46,35 @@ RUN for i in $(seq 1 $REPO_COUNT); do \
     chmod +x "/var/www/git/repo$i.git/hooks/post-receive"; \
     done
 
+#creating repositories which donot require authentication and can be accessed from http://localhost:8080/gitopen
```
Comment (juliev0) [inline tests/gitserver/Dockerfile:49]:
```suggestion
#creating repositories which do not require authentication and can be accessed from http://localhost:8080/gitopen
```

File: tests/gitserver/Dockerfile [hunk 3]
```diff
@@ -94,13 +114,29 @@ RUN for i in $(seq 1 $REPO_COUNT); do \
...
-    git add data.yaml && \
+    echo "Local Repo" > readme.md && \
+    git add readme.md && \
     git commit -m "Initial commit" && \
     git push /var/www/git/repo$i.git master;\
     done
 #to access http://localhost:8080/git/repo2.git http://localhost:8080/git/repo1.git
 
+
+# Creating a non bare git repo  which donot require  authentication with working directory as per the RepoCount variable and pushing the changes
```
Comment (juliev0) [inline tests/gitserver/Dockerfile:125]:
```suggestion
# Creating a non bare git repo  which doesn't require authentication with working directory as per the RepoCount variable and pushing the changes
```

**Pull request merged**